rosetta/retriever/retriever.go

```diff
@@ -262,6 +263,30 @@ func (r *Retriever) Transaction(block identifier.Block, id identifier.Transactio
 		return nil, fmt.Errorf("could not validate transaction: %w", err)
 	}
 
+	blockID, err := flow.HexStringToIdentifier(completed.Hash)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert block ID: %w", err)
+	}
+
+	transactions, err := r.index.Transactions(blockID)
+	if err != nil {
+		return nil, fmt.Errorf("could not list block transactions: %w", err)
+	}
+
+	var found bool
+	for _, txID := range transactions {
+		if txID.String() == id.Hash {
+			found = true
```

Comment:
Since the transaction ID is unique, we can probably break out of the loop here..?